tests/beaker_tests/lib/utilitylib.rb

```diff
@@ -213,10 +221,25 @@ def test_manifest(tests, id)
   stepinfo = format_stepinfo(tests, id, 'MANIFEST')
   step "TestStep :: #{stepinfo}" do
     logger.debug("test_manifest :: manifest:\n#{tests[id][:manifest]}")
-    on(tests[:master], tests[id][:manifest])
-    code = tests[id][:code] ? tests[id][:code] : [2]
-    logger.debug("test_manifest :: check puppet agent cmd (code: #{code})")
-    on(tests[:agent], puppet_agent_cmd, acceptable_exit_codes: code)
+    if tests[:master]
+      on(tests[:master], tests[id][:manifest])
+      code = (tests[id][:code]) ? tests[id][:code] : [2]
+      logger.debug("test_manifest :: check puppet agent cmd (code: #{code})")
+      on(tests[:agent], puppet_agent_cmd, acceptable_exit_codes: code)
+    else
+      system(tests[id][:manifest])
+      code = (tests[id][:code]) ? tests[id][:code] : [2]
+      logger.debug("test_manifest :: check puppet apply cmd (code: #{code})")
+      # system("bundle exec puppet device --verbose --trace --strict=error --modulepath spec/fixtures --target nexus --libdir lib/ --apply apply.pp")
+      # See https://tickets.puppetlabs.com/browse/PUP-9067 "`puppet device` should respect --detailed-exitcodes"
+      # if !code.include?($?.exitstatus)
+      #  raise 'Errored test'
+      # end
+      output = `bundle exec puppet device --verbose --trace --strict=error --modulepath spec/fixtures --target nexus --libdir lib/ --apply apply.pp`
```

Comment:
you may need to ensure that `bundle exec rake spec_prep` has been ran before this

Reply:
the tests themselves chould be running in an appropriate environment, which is already `spec_prep`ped, and should not require a nested `bundle` call.